src/query/functions/temporal/aggregation.go

```diff
@@ -90,78 +97,67 @@ func (a *aggNode) Process(values []float64) float64 {
 }
 
 func avgOverTime(values []float64) float64 {
-	var sum, count float64
-	for _, v := range values {
-		if !math.IsNaN(v) {
-			count++
-			sum += v
-		}
-	}
-
+	sum, count := sumAndCount(values)
 	if count == 0 {
 		return math.NaN()
 	}
+
 	return sum / count
 }
 
 func countOverTime(values []float64) float64 {
-	var count float64
-	for _, v := range values {
-		if !math.IsNaN(v) {
-			count++
-		}
-	}
+	_, count := sumAndCount(values)
 	return count
 }
 
 func minOverTime(values []float64) float64 {
-	var count float64
+	var seenNotNaN bool
 	min := math.Inf(1)
 	for _, v := range values {
 		if !math.IsNaN(v) {
-			count++
+			seenNotNaN = true
 			min = math.Min(min, v)
 		}
 	}
 
-	if count == 0 {
+	if !seenNotNaN {
 		return math.NaN()
 	}
+
 	return min
 }
 
 func maxOverTime(values []float64) float64 {
-	var count float64
+	var seenNotNaN bool
 	max := math.Inf(-1)
 	for _, v := range values {
 		if !math.IsNaN(v) {
-			count++
+			seenNotNaN = true
 			max = math.Max(max, v)
 		}
 	}
 
-	if count == 0 {
+	if !seenNotNaN {
 		return math.NaN()
 	}
+
 	return max
 }
 
 func sumOverTime(values []float64) float64 {
-	var sum, count float64
-	for _, v := range values {
-		if !math.IsNaN(v) {
-			count++
-			sum += v
-		}
-	}
-
+	sum, count := sumAndCount(values)
 	if count == 0 {
 		return math.NaN()
```

Comment:
nit: this logic might be better to put in sumAndCount?

Reply:
But then you'd still have to have an if statement here, no?